modules/core/src/main/java/org/apache/ignite/internal/processors/metric/export/MetricResponse.java

```diff
@@ -0,0 +1,484 @@
+            long ts,
+            UUID clusterId,
+            byte[] userTagBytes,
+            byte[] consistentIdBytes,
+            int schemaOff,
+            int schemaSize,
+            int dataOff,
+            int dataSize
+    ) {
+        putInt(body, BYTE_ARR_OFF + MSG_SIZE_OFF, body.length);
+
+        putShort(body, BYTE_ARR_OFF + PROTO_VER_OFF, PROTO_VER_1);
+
+        putLong(body, BYTE_ARR_OFF + TIMESTAMP_OFF, ts);
+
+        putInt(body, BYTE_ARR_OFF + SCHEMA_VER_OFF, schemaVer);
+
+        putInt(body, BYTE_ARR_OFF + SCHEMA_OFF_OFF, schemaOff);
+
+        putInt(body, BYTE_ARR_OFF + SCHEMA_SIZE_OFF, schemaSize);
+
+        putInt(body, BYTE_ARR_OFF + DATA_OFF_OFF, dataOff);
+
+        putInt(body, BYTE_ARR_OFF + DATA_SIZE_OFF, dataSize);
+
+        putLong(body, BYTE_ARR_OFF + CLUSTER_ID_OFF, clusterId.getMostSignificantBits());
+
+        putLong(body, BYTE_ARR_OFF + CLUSTER_ID_OFF + Long.BYTES, clusterId.getLeastSignificantBits());
+
+        int off = BASE_HEADER_SIZE;
+
+        if (userTagBytes != null) {
+            putInt(body, BYTE_ARR_OFF + USER_TAG_SIZE_OFF, userTagBytes.length);
+
+            copyMemory(userTagBytes, BYTE_ARR_OFF, body, BYTE_ARR_OFF + off, userTagBytes.length);
+
+            off += userTagBytes.length;
+        }
+        else
+            putInt(body, BYTE_ARR_OFF + USER_TAG_SIZE_OFF, 0);
+
+        putInt(body, BYTE_ARR_OFF + off, consistentIdBytes.length);
+
+        off += Integer.BYTES;
+
+        copyMemory(consistentIdBytes, BYTE_ARR_OFF, body, BYTE_ARR_OFF + off, consistentIdBytes.length);
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean writeTo(ByteBuffer buf, MessageWriter writer) {
+        writer.setBuffer(buf);
+
+        if (!writer.isHeaderWritten()) {
+            if (!writer.writeHeader(directType(), fieldsCount()))
+                return false;
+
+            writer.onHeaderWritten();
+        }
+
+        switch (writer.state()) {
+            case 0:
+                if (!writer.writeByteArray("body", body))
+                    return false;
+
+                writer.incrementState();
+        }
+
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean readFrom(ByteBuffer buf, MessageReader reader) {
+        reader.setBuffer(buf);
+
+        if (!reader.beforeMessageRead())
+            return false;
+
+        switch (reader.state()) {
+            case 0:
+                body = reader.readByteArray("body");
+
+                if (!reader.isLastRead())
+                    return false;
+
+                reader.incrementState();
+        }
+
+        return reader.afterMessageRead(MetricResponse.class);
+    }
+
+    /** {@inheritDoc} */
+    @Override public short directType() {
+        return -62;
+    }
+
+    /** {@inheritDoc} */
+    @Override public byte fieldsCount() {
+        return 1;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onAckReceived() {
+        // No-op.
+    }
+}
```

Comment:
Do we need `toString()` method for this class?

Reply:
Added toString()